### PR TITLE
feat: [CDB-178]: only retrieve name when viewing custom folders

### PR DESCRIPTION
**Pull request by sean-harness**
##### Summary:

Avoid making an unnecessary API call when viewing the 'shared' folder.

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by Harness-JKhattak** [comment]
retrigger jest

**Review by Harness-JKhattak — DISMISSED** [review]

**Review by Harness-JKhattak — APPROVED** [review]

File: src/modules/45-dashboards/components/DashboardResourceModalBody/DashbaordResourceModalBody.tsx [hunk 1]
```diff
@@ -113,9 +111,9 @@ const DashboardResourceModalBody: React.FC<RbacResourceModalProps> = ({
           }
         ]}
         pagination={{
-          itemCount: totalFolders || 1,
+          itemCount: folders?.items || 1,
```
Comment (knagurski) [inline src/modules/45-dashboards/components/DashboardResourceModalBody/DashbaordResourceModalBody.tsx:114]: Do you want to default to `1`? Seems like if the thing is empty, you probably don't have any rows to show.

**Review by knagurski — APPROVED** [review]
Looks good, but would be better with a few tests

**Comment by sean-harness** [comment]
retrigger jest

**Comment by sean-harness** [comment]
retrigger eslint

**Comment by sean-harness** [comment]
retrigger jest

File: src/modules/45-dashboards/components/DashboardResourceModalBody/DashbaordResourceModalBody.tsx (hunk 1 above)
Comment (sean-harness) [reply src/modules/45-dashboards/components/DashboardResourceModalBody/DashbaordResourceModalBody.tsx]: Good point, changed to 0

**Review by Harness-JKhattak — APPROVED** [review]
Looks good to me.

The Coverage check might be failing but my PR will be remedying this by splitting out the components in HomePage and adding a lot of test coverage to the new components.

File: src/modules/45-dashboards/pages/home/HomePage.tsx [hunk 2]
```diff
@@ -125,13 +125,16 @@ const TagsRenderer = (data: DashboardInterface) => {
...
+        data?.description
+          .split(',')
+          .filter((tag: string) => tag !== '')
+          .map((tag: string, index: number) => {
+            return (
+              <section className={css.customTag} key={tag + index}>
+                {tag}
+              </section>
+            )
+          })}
```
Comment (knagurski) [inline src/modules/45-dashboards/pages/home/HomePage.tsx:137]: Absolutely nothing wrong with this `map`, but you can drop a few characters for things like this with an implicit return.
```suggestion
          .map((tag: string, index: number) => (
            <section className={css.customTag} key={tag + index}>
              {tag}
            </section>
          ))}
```

**Review by knagurski — DISMISSED** [review]
Looks good, just a few optional opinions/suggestions and one filename typo.

**Review by Harness-JKhattak — DISMISSED** [review]

**Comment by sean-harness** [comment]
retrigger jest

**Comment by sean-harness** [comment]
retrigger jest

**Comment by sean-harness** [comment]
retrigger titlecheck

**Comment by sean-harness** [comment]
retrigger jest

File: src/modules/45-dashboards/pages/home/__tests__/HomePage.test.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  test('it should show an empty message when there are no dashboards', () => {
+    renderComponent()
+
+    expect(screen.getByText('dashboardLabel')).toBeInTheDocument()
+  })
+
+  test('it should change the sort order when a sort option is clicked', async () => {
+    renderComponent()
+
+    const sortByButton = screen.getByText('dashboards.sortBy Select Option') as HTMLButtonElement
```
Comment (knagurski) [inline src/modules/45-dashboards/pages/home/__tests__/HomePage.test.tsx:55]: For buttons (links, list items etc), try to lean towards `getByRole`
```suggestion
    const sortByButton = screen.getByRole('button', { name: 'dashboards.sortBy Select Option' })
```

File: src/modules/45-dashboards/pages/home/__tests__/HomePage.test.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    await act(async () => {
+      userEvent.click(sortByButton)
+    })
```
Comment (knagurski) [inline src/modules/45-dashboards/pages/home/__tests__/HomePage.test.tsx:58]: You don't need to wrap `userEvent` calls in `act`; they're already wrapped for you.
```suggestion

    userEvent.click(sortByButton)
    
```

File: src/modules/45-dashboards/types/DashboardTypes.tsx [hunk 5]
```diff
@@ -5,19 +5,6 @@
...
-  title: string
-  view_count: number
-  favorite_count: number
-  created_at: string
-  data_source: DashboardTag[] | string[]
-  last_accessed_at: string
-  resourceIdentifier: string
-}
-
 export enum DashboardTag {
```
Comment (knagurski) [inline src/modules/45-dashboards/types/DashboardTypes.tsx:8]: Is this still used?

**Review by knagurski — DISMISSED** [review]
Just a few suggestions, but looks good!

File: src/modules/45-dashboards/components/DashboardResourceModalBody/DashboardResourceModalBody.tsx [hunk 6]
```diff
@@ -84,18 +89,15 @@ const DashboardResourceModalBody: React.FC<RbacResourceModalProps> = ({
   const [page, setPage] = React.useState(0)
   const { getString } = useStrings()
 
-  const { data: folders, loading: fethingFolders } = useGet({
-    // Inferred from RestfulProvider in index.js
-    path: 'gateway/dashboard/folder',
+  const { data: folders, loading: fethingFolders } = useGetFolder({
```
Comment (Harness-JKhattak) [inline src/modules/45-dashboards/components/DashboardResourceModalBody/DashboardResourceModalBody.tsx:92]: typo in "fethingFolders"

```suggestion
  const { data: folders, loading: fetchingFolders } = useGetFolder({
```

**Review by Harness-JKhattak — APPROVED** [review]
good to see test coverage passing in HomePage. We'll be top of the coverage leaderboards soon 😛

**Review by Harness-JKhattak — DISMISSED** [review]

**Comment by sean-harness** [comment]
retrigger sonar

**Comment by sean-harness** [comment]
retrigger jest

**Comment by sean-harness** [comment]
trigger cypress

**Comment by sean-harness** [comment]
retrigger sonar

**Review by Harness-JKhattak — APPROVED** [review]

**Comment by sean-harness** [comment]
trigger cypress

**Comment by sean-harness** [comment]
retrigger sonar

**Comment by sean-harness** [comment]
retrigger sonar

**Comment by sean-harness** [comment]
retrigger sonar

**Review by Harness-JKhattak — APPROVED** [review]
Good to see us nearly fully away from using the module specific Service file 👏

**Review by knagurski — APPROVED** [review]
Nice!

**Pull request merged**